src/new-components/banner/banner.tsx

```diff
@@ -0,0 +1,80 @@
+import * as React from 'react'
+import { Box } from '../box'
+import { Columns, Column } from '../columns'
+
+import styles from './banner.module.css'
+
+export type BannerTone = 'info' | 'promotion'
+
+type BannerProps = {
+    id?: string
+    /**
+     * The tone of the Banner. Affects the background color and the outline.
+     */
+    tone: BannerTone
+    /**
+     * The icon that should be added inside the Banner.
+     */
+    icon: React.ReactChild
+    /**
+     * The title to be displayed at the top of the Banner.
+     */
+    title: string
+    /**
+     * An optional description to be displayed inside the Banner.
+     */
+    description?: string
+    /**
+     * An optional action to displayed inside the Banner.
+     */
+    action?: React.ReactChild
+}
+
+function Banner({ id, tone, icon, title, description, action, ...props }: BannerProps) {
+    return (
+        <Box
+            {...props}
+            id={id}
+            role="banner"
```

Comment:
A note: if I set this `role="banner"` and I try to render more than one banner at the time (i.e. one stacked top of each other) I see that the a11y doesn't pass:

```
● Banner › renders with no a11y violations

    expect(received).toHaveNoViolations(expected)

    Expected the HTML found at $('.banner-info') to have no violations:

    <div role="banner" aria-live="polite" class="banner banner-info box borderRadius-standard">

    Received:

    "Document should not have more than one banner landmark (landmark-no-duplicate-banner)"

    Fix any of the following:
      Document has more than one banner landmark

    You can find more information on this issue here:
    https://dequeuniversity.com/rules/axe/4.2/landmark-no-duplicate-banner?application=axeAPI

    ────────

    Expected the HTML found at $('.banner-info') to have no violations:

    <div role="banner" aria-live="polite" class="banner banner-info box borderRadius-standard">

    Received:

    "Ensures landmarks are unique (landmark-unique)"

    Fix any of the following:
      The landmark must have a unique aria-label, aria-labelledby, or title to make landmarks distinguishable

    You can find more information on this issue here:
    https://dequeuniversity.com/rules/axe/4.2/landmark-unique?application=axeAPI

      79 |         )
      80 |         const results = await axe(container)
    > 81 |         expect(results).toHaveNoViolations()
         |                         ^
      82 |     })
      83 | })
      84 |

      at Object.toHaveNoViolations (src/new-components/banner/banner.test.tsx:81:25)
```

More info at this link: https://dequeuniversity.com/rules/axe/4.2/landmark-no-duplicate-banner?application=axeAPI

> The role="banner" ARIA landmark should only appear once on an element that is unique to the HTML document

Reply:
Wow, it's good that we have these jest axe tests in place.

Indeed, it seems this component should not use `role="banner"`. In spite of our choice of name, in terms of aria roles, "banner" seems to something else entirely. According to [this](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/Banner_Role):

> The `banner` role is for defining a global site header, which usually includes a logo, company name, search feature, and possibly the global navigation or a slogan. It is generally located at the top of the page.

So it's nothing like what you are developing here.

Looking at the list of existing aria roles, I think this component is in line with [`role="note"`](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/note_role). Check that out, and if you agree, make the change.

Reply:
@gnapse Thank you for all the context! 🙏 I think `role="note"` is perfect, as this is just something slightly more prominent, that we want the users to pay attention to, but nothing more.

Does this in any way suggest that we should rename our new reactist component to something else? I felt like `Banner` worked well, but I also feel like I don't know English enough maybe.

Reply:
Let's ask @craigcarlyle, not only as a native English speaker, but as our accessibility DRD.

Reply:
First of all, I'm happy to see that the automated tests caught this 🎉 

I think setting the `role` to `note` is okay, but thinking about our use cases maybe we should follow the example of the [Banner component in the Polaris design system](https://polaris.shopify.com/components/banner) and: 

* Set the `role` to `status`
* Set the `aria-live` property to `polite`
* Add `tabindex="0"` (so that it can be navigated to with a keyboard)

This means that users with screen reading software would:
* Get notified eventually that they're viewing an achieved project
* Get notified eventually that they're previewing a project
* Get notified eventually that they've hit a project limit

If we went with the `note` option the user would have to manually navigate to the content to discover it instead of the content being announced automatically. I think it may be the better option.

And to answer the last question, I think the "Banner" name is fine. The name is for developers rather than the end consumer 🙂

Reply:
@craigcarlyle, name-similarities aside, it does not look like Polaris' component is meant for the same use cases as ours.

- That one looks more like something that appears triggered by some user action (judging by that default example of "Order archived") and it is dismissable.
- Ours is meant to appear from the start if the conditions for it are met, and it is not dismissable, but stays around. At least not dismissable by clicking a close button. The only way you can dismiss a "This project is preview only" is to click on "Join project", a permanent action that removes the condition that made the banner appear.

I'm not saying these are reasons good enough to not use `role="status"`, but I do see the difference with what, at a glance, the Polaris' component seems to be about. With all this in mind, do you still think we should use that option? The part about this message being announced might be interesting. But, again, this message does not suddenly appear due to a user action.

Reply:
Actually, that Polaris component seems more like Reactist's `Alert`, a component that is visually similar to this one, but with a different semantics. Check out [this comment](https://github.com/Doist/Issues/issues/7617#issuecomment-1397364983) where I proposed to make these different things in the design system.

Reply:
@gnapse I still believe it's the better option and I think it's differentiated between the `Alert` component enough with the role difference (`status` vs `alert`). However, I don't feel too strongly about it.

Looking at Polaris closer, our use case may be closer to the [Callout card](https://polaris.shopify.com/components/callout-card) component which has no special `role` or `aria-*` properties at all. Maybe we're overthinking it? 🤷‍♂️

Reply:
Yes, viewed under that light I see your point. Comparing `role="status"` vs. `role="alert"`, I see this about `status`:

> The `status` role defines a [live region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) containing advisory information for the user that is not important enough to be an [alert](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/alert_role).

[source](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/status_role)

Thanks!

What do you think @engfragui? Do we have a winner? 😄

Reply:
Thank you both for this discussion! 🙏 Let's go with role "status", which feels like in between "alert" and "note" and might fit well with our intent of "we want the user to pay attention to this, but not really alert them".

Just to confirm, this is what I have:

```
            role="status"
            aria-labelledby={titleId}
            aria-describedby={descriptionId}
            aria-live="polite"
            tabIndex={0}
```